crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -34,6 +34,30 @@ pub struct Circle {
 }
 impl Primitive2d for Circle {}
 
+impl Circle {
+    /// Create a `Circle` from a radius
+    pub fn new(radius: f32) -> Self {
+        Self { radius }
+    }
+}
+
+/// An ellipse primitive
+#[derive(Clone, Copy, Debug)]
+pub struct Ellipse {
+    /// The "width" of the ellipse
+    pub width: f32,
+    /// The "height" of the ellipse
+    pub height: f32,
```

Comment:
Maybe half-width and half-height could be more efficient like for other shapes? Not sure for ellipses

Reply:
Hmm. The main operations would be:

- draw the curve
- area: half-extents
- perimeter: half-extents, approximate only
- containment: half-extents

Definitely think we should swap.